src/Symfony/Component/Messenger/EventListener/StopWorkerOnSigtermSignalListener.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Messenger\Event\WorkerStartedEvent;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ */
+class StopWorkerOnSigtermSignalListener implements EventSubscriberInterface
+{
+    public function onWorkerStarted(WorkerStartedEvent $event): void
+    {
+        pcntl_signal(SIGTERM, function () use ($event) {
+            $event->getWorker()->stop();
+        });
+    }
+
+    public static function getSubscribedEvents()
+    {
+        if (!\function_exists('pcntl_signal')) {
```

Comment:
This is essentially a silent failure when the extension is not installed. Wouldn't you want to throw an exception here? Otherwise this could not be working as intended, which is unexpected.